src/Paprika/Data/SlottedArray.cs

```diff
@@ -177,25 +175,25 @@ public bool MoveNext()
             if (index < to)
             {
                 _index = index;
-                _current = Build();
+                Build(out _current);
                 return true;
             }
 
             return false;
         }
 
-        public Item Current => _current;
+        public readonly Item Current => _current;
 
-        private Item Build()
+        private void Build(out Item value)
```

Comment:
Would be great if you could add an enumerator benchmark to `SlottedArrayBenchmarks` to cover this and other changes.